lib/types/types.ts

```diff
@@ -128,17 +133,100 @@ export interface ColumnData {
     file_path: string,
     file_offset: Offset,
     meta_data: ColumnMetaData
+    offset_index_length?: number;
+    column_index_length?: number;
+    encrypted_column_metadata?: Buffer;
+    offsetIndex?: OffsetIndex;
+    offset_index_offset?: number;
+    columnIndex?: ColumnIndex;
+    column_index_offset?: number;
 }
 
 export interface ColumnChunkData {
     rowGroupIndex: number,
     column: ColumnData
 }
 
+export declare class RowGroup {
+    columns: ColumnData[];
+    num_rows: number;
+    ordinal?: number;
+}
+
+export declare class KeyValue {
+    key: string;
+    value?: string;
+}
+
 export type Block = Uint32Array
 
 export interface BloomFilterData {
     sbbf: SplitBlockBloomFilter,
     columnName: string,
     RowGroupIndex: number,
 };
+
+export interface Parameter {
+    url: string;
+    headers?: string
+}
+
+export interface PageData {
+    rlevels?: number[];
+    dlevels?: number[];
+    values?: number[];
+    pageHeaders?: PageHeader[];
+    pageHeader?: PageHeader;
+    count?: number;
+    dictionary?: Array<unknown>
+}
+
+export declare class PageHeader {
+    type: PageType;
+    uncompressed_page_size: number;
+    compressed_page_size: number;
+    crc?: number;
+    data_page_header?: DataPageHeader;
+    index_page_header?: IndexPageHeader;
+    dictionary_page_header?: DictionaryPageHeader;
+    data_page_header_v2?: DataPageHeaderV2;
+    offset?: number;
+    headerSize?: number;
+  
+      constructor(args?: { type: PageType; uncompressed_page_size: number; compressed_page_size: number; crc?: number; data_page_header?: DataPageHeader; index_page_header?: IndexPageHeader; dictionary_page_header?: DictionaryPageHeader; data_page_header_v2?: DataPageHeaderV2; });
+  }
+
+  export interface ClientParameters {
+    Bucket: string,
+    Key: string
+  }
+
+  export interface PromiseS3 {
+      promise: () => Promise<any>
+  }
+
+  export interface ClientS3 {
+    accessKeyId: string,
+    secretAccessKey: string,
+    headObject: (params: ClientParameters) => PromiseS3
+    getObject: (args: any) => PromiseS3
+}
+
+export class NewFileMetaData extends parquet_thrift.FileMetaData {
+    json?:JSON;
+    //@ts-ignore
+    row_groups:RowGroup[];
+    constructor() {
+      super()
+    } 
+  }
```

Comment:
This is interesting because the code requires some of the objects be thrift objects with the built in functions, but some calls to the object that is made asks for values that don't exist on the original thrift object which is why I created a new version of that object to add and  overwrite some of the typing